lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -533,10 +541,46 @@ def get_object_ref(self, module, ib_obj_type, obj_filter, ib_spec):
                 # resolves issue where a_record with uppercase name was returning null and was failing
                 test_obj_filter = obj_filter
                 test_obj_filter['name'] = test_obj_filter['name'].lower()
+                # resolves issue where multiple a_records with same name and different IP address
+                try:
+                    ipaddr_obj = self.module._check_type_dict(obj_filter['ipv4addr'])
+                    ipaddr = ipaddr_obj['old_ipv4addr']
+                except TypeError:
+                    ipaddr = obj_filter['ipv4addr']
+                test_obj_filter['ipv4addr'] = ipaddr
+            elif (ib_obj_type == NIOS_TXT_RECORD):
```

Comment:
It is repeated twice since the first condition is to check in the following scenario:
- When old_name, new_name, old_text and new_text parameters are supplied from the playbook for modification.
The second scenario is implemented in the following scenario:
- When only old_text and new_text is supplied for modification requests